lib/admin_credentials.rb

```diff
@@ -19,7 +19,21 @@ def initialize(env, **opts)
 
     # @return [APIAccessor]
     def get
-      res = master_host.exec_as(:admin, "cat /root/.kube/config", quiet: true)
+      res = master_host.exec_admin("cat /root/.kube/config", quiet: true)
+      if !res[:success] && res[:response].include?("No such file or directory")
```

Comment:
@akostadinov Hmm, today encountered [another failure](http://ci-qe-openshift.usersys.redhat.com/userContent/cucushift/v3/2019/01/24/00:50:14/Create_service_with_multiports/console.html) due to non-robust code here:
```
undefined method `unpack' for nil:NilClass
NoMethodError
/opt/rh/rh-ruby24/root/usr/share/ruby/base64.rb:59:in `decode64'
/home/jenkins/workspace/Runner-v3/lib/admin_credentials.rb:54:in `get'
/home/jenkins/workspace/Runner-v3/lib/environment.rb:89:in `admin'
/home/jenkins/workspace/Runner-v3/lib/environment.rb:333:in `nodes'
/home/jenkins/workspace/Runner-v3/features/step_definitions/node.rb:11:in `/^I select a random node's host$/'
features/tierN/cli/service.feature:125:in `And I select a random node's host'
```
I debugged on the master: http://pastebin.test.redhat.com/701353 , its /root/.kube/config is just the one shown in the failure log's output, which does not include admin cert/key. The failure cause is somebody did oc login on master with normal user.
Could you help fix it to work more robust even when somebody did normal user oc login on master?
BTW, I'll also suggest team try to not do normal user oc login on master.

Reply:
We can make it work with the token from the file but token is temporary and next day it will not work. Not sure it is a good idea.